From ab1763a5c1c952c4a77f29cdf9302cfa2beab4e2 Mon Sep 17 00:00:00 2001
From: Su Bao Cheng <baocheng.su@siemens.com>
Date: Mon, 22 Nov 2021 15:59:01 +0800
Subject: [PATCH 117/118] Revert "serial: 8250: Don't touch RTS modem control
 while in rs485 mode"

This reverts commit f45709df7731ad36306a28a3e1af7309d55c35f5.

The mcr register is not always reflect the significant value, so
checking against it will mislead.

The direct issue is RS485/422 does not fully work on 8250 serial, the
root cause is that the original patch added a fucntion to avoid user
space trigged changes but this change incorrectly blocked also valid
kernel-originated changes because it hooked into the wrong function.

Details refer to the bottom link.

Aleady reported to upstream and Lukas is working on a formal fix and
after that is accepted by upstream, this reverting will be replace by
the formal fix.

Signed-off-by: Su Bao Cheng <baocheng.su@siemens.com>
Link: https://lore.kernel.org/r/20211027111644.1996921-1-baocheng.su@siemens.com
---
 drivers/tty/serial/8250/8250_port.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/drivers/tty/serial/8250/8250_port.c b/drivers/tty/serial/8250/8250_port.c
index 3de0a16e055a..37e859da7d60 100644
--- a/drivers/tty/serial/8250/8250_port.c
+++ b/drivers/tty/serial/8250/8250_port.c
@@ -2028,13 +2028,6 @@ void serial8250_do_set_mctrl(struct uart_port *port, unsigned int mctrl)
 	struct uart_8250_port *up = up_to_u8250p(port);
 	unsigned char mcr;
 
-	if (port->rs485.flags & SER_RS485_ENABLED) {
-		if (serial8250_in_MCR(up) & UART_MCR_RTS)
-			mctrl |= TIOCM_RTS;
-		else
-			mctrl &= ~TIOCM_RTS;
-	}
-
 	mcr = serial8250_TIOCM_to_MCR(mctrl);
 
 	mcr = (mcr & up->mcr_mask) | up->mcr_force | up->mcr;
-- 
2.33.1

